### PR TITLE
MHC being more tolerant  from 1 to 33%

### PR DESCRIPTION
**Pull request by ehvs**
### Which issue this PR addresses:

https://redhat.atlassian.net/browse/ARO-27497

### What this PR does / why we need it:

Make it more tolerant over worker nodes issues.

This PR adjusts the default OpenShift MachineHealthCheck configuration deployed by the ARO operator to be more tolerant of worker node health issues by switching maxUnhealthy from a fixed count to a percentage threshold.

Changes:
Update MachineHealthCheck.spec.maxUnhealthy from "1" to "33%" to scale the unhealthy tolerance with cluster size.

### Test plan for issue:

e2e and existent unit tests 
### Is there any documentation that needs to be updated for this PR?
No

### CODE REVIEW

**Review by shubhadapaithankar — COMMENTED** [review]
Looks good to me! A couple of optional suggestions:

1. The unit tests currently don't validate the actual MHC spec contents (they only check the paused annotation). Might be worth adding a small test that parses the embedded YAML and asserts `maxUnhealthy == "33%"` to prevent accidental regressions - could be a follow-up though.
2. One edge case worth noting: on a 2-worker cluster, 33% rounds down to 0, which would actually be more restrictive than the current `"1"`. Since ARO enforces a 3-worker minimum this shouldn't be an issue in practice, just calling it out.

**Review by alcasim — APPROVED** [review]
LGTM

I wonder if you had the chance to test this change locally, it should work as expected, also it should reduce the number of occurrences of the MHC short circuit alert.

File: pkg/operator/controllers/machinehealthcheck/staticresources/machinehealthcheck.yaml [hunk 1]
```diff
@@ -20,5 +20,5 @@ spec:
   - type: "Ready"
     timeout: "15m"
     status: "Unknown"
-  maxUnhealthy: "1"
+  maxUnhealthy: "33%"
```
Comment (swiencki) [inline pkg/operator/controllers/machinehealthcheck/staticresources/machinehealthcheck.yaml:23]: OpenShift MHC uses `math.Floor` for percentage values ([source](https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go#L416)). For a 3-worker pool that gives `floor(33 * 3 / 100) = 0`, which would block all remediation. `"34%"` would keep `floor(1.02) = 1`, same as today for 3-node clusters while still scaling up.

Comment (swiencki) [reply pkg/operator/controllers/machinehealthcheck/staticresources/machinehealthcheck.yaml:23]: I also wonder how this would now compare to ROSA

**Comment by swiencki** [comment]
> 2. One edge case worth noting: on a 2-worker cluster, 33% rounds down to 0, which would actually be more restrictive than the current `"1"`. Since ARO enforces a 3-worker minimum this shouldn't be an issue in practice, just calling it out.

A two node edge case that will apply is infra nodes. That is planning to be added in [PR 4832](https://github.com/Azure/ARO-RP/pull/4832).
